lib/web/terminal.go

```diff
@@ -161,6 +161,9 @@ type TerminalHandler struct {
 
 	// decoder is used to decode UTF-8 strings.
 	decoder *encoding.Decoder
+
+	// read buffer
```

Comment:
Use this comment to explain why `buffer` is needed: the payload can contain more data than the size of the caller to `Read` provides in the buffer to fill out.